doc/rados/configuration/bluestore-config-ref.rst

```diff
@@ -369,167 +373,173 @@ throttling
-as one 4kb and one 1kb rados object, again stranding 4kb of device capacity,
-though in this case the overhead is a much smaller percentage.  think of this
-in terms of the remainder from a modulus operation. the overhead *percentage*
-thus decreases rapidly as user object size increases.
-
-an easily missed additional subtlety is that this
-takes place for *each* replica.  so when using the default three copies of
-data (3r), a 1kb s3 object actually consumes 12kb of storage device
-capacity, with 11kb of overhead.  if erasure coding (ec) is used instead of replication, the
-amplification may be even higher: for a ``k=4,m=2`` pool, our 1kb s3 object
-will allocate (6 * 4kb) = 24kb of device capacity.
-
-when an rgw bucket pool contains many relatively large user objects, the effect
-of this phenomenon is often negligible, but should be considered for deployments
-that expect a significant fraction of relatively small objects.
-
-the 4kb default value aligns well with conventional hdd and ssd devices.  some
-new coarse-iu (indirection unit) qlc ssds however perform and wear best
-when :confval:`bluestore_min_alloc_size_ssd`
-is set at osd creation to match the device's iu:. 8kb, 16kb, or even 64kb.
-these novel storage drives allow one to achieve read performance competitive
-with conventional tlc ssds and write performance faster than hdds, with
-high density and lower cost than tlc ssds.
-
-note that when creating osds on these devices, one must carefully apply the
-non-default value only to appropriate devices, and not to conventional ssd and
-hdd devices.  this may be done through careful ordering of osd creation, custom
-osd device classes, and especially by the use of central configuration _masks_.
-
-quincy and later releases add
-the :confval:`bluestore_use_optimal_io_size_for_min_alloc_size`
-option that enables automatic discovery of the appropriate value as each osd is
-created.  note that the use of ``bcache``, ``opencas``, ``dmcrypt``,
-``ata over ethernet``, `iscsi`, or other device layering / abstraction
-technologies may confound the determination of appropriate values. osds
-deployed on top of vmware storage have been reported to also
-sometimes report a ``rotational`` attribute that does not match the underlying
-hardware.
-
-we suggest inspecting such osds at startup via logs and admin sockets to ensure that
-behavior is appropriate.  note that this also may not work as desired with
-older kernels.  you can check for this by examining the presence and value
-of ``/sys/block/<drive>/queue/optimal_io_size``.
-
-you may also inspect a given osd:
+If these settings are not entered, then the current implementation will
+populate the SPDK map files with kernel file system symbols and will use the
+kernel driver to issue DB/WAL I/Os.
+
+Minimum Allocation Size
+=======================
+
+There is a configured minimum amount of storage that BlueStore allocates on an
```

Comment:
s/on an OSD/on an underlying storage device/
BlueStore is part of the OSD, so as-is this is self-referential.

Reply:
Accepted.